src/client/builder.rs

```diff
@@ -183,7 +183,8 @@ impl HttpRequestBuilder {
         let mut error = None;
         if let Ok(ref mut req) = self.request {
             let mut out = format!("{}?", req.uri().path());
-            let mut encoder = form_urlencoded::Serializer::new(&mut out);
+            let start_position = out.len();
```

Comment:
> [apache/arrow-rs#7183](https://github.com/apache/arrow-rs/pull/7183) Changed the way the query string is serialized from:
> 
> ```
> azurebase.com/path?comp=list&restype:container
> ```
> 
> to:
> 
> ```
> azurebase.com/path?&comp=list&restype:container # Note the leading & after the ?
> ```

Does this change remove the  `&` after `?`. I am having a hard time reasoning about the change this behavior makes. 

Is it possible to write some unit tests or something that can show the before/after behavior (and prevent this code from being accidentally broken again)?

Reply:
yeah, after this change the URL will again be `azurebase.com/path?comp=list&restype:container`. 

Part of the problem here is that I realized that it seems like [Azurite behaves differently](https://github.com/Azure/Azurite/issues/2548) than Azure in this case, so tests that will fail on the real Azure (like @a10y's original repro) would pass in the repo's test suite.

Reply:
Maybe a unit test then?